ldpd/lde.c

```diff
@@ -102,9 +115,31 @@ static struct quagga_signal_t lde_signals[] =
 	},
 };
 
+struct zclient *zclient_sync = NULL;
+static void
+zclient_sync_init(u_short instance)
+{
+	/* Initialize special zclient for synchronous message exchanges. */
+	log_debug("Initializing synchronous zclient for label manager");
+	zclient_sync = zclient_new(master);
+	zclient_sync->sock = -1;
+	zclient_sync->redist_default = ZEBRA_ROUTE_LDP;
+	zclient_sync->instance = instance;
+	while (zclient_socket_connect(zclient_sync) < 0) {
+		fprintf(stderr, "Error connecting synchronous zclient!\n");
+		sleep(1);
+	}
+
+	/* Connect to label manager */
+	while (lm_label_manager_connect(zclient_sync) != 0) {
+		fprintf(stderr, "Error connecting to label manager!\n");
+		sleep(1);
+	}
+}
+
 /* label decision engine */
```

Comment:
What's happen if zclient_socket_connect or lm_label_manager_connect never success ? The daemon loop indefinitely ? Perhaps, it will be safer to try 3 or 10 times and then exit if connection to Label Manager is not possible.